ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.hpp

```diff
@@ -19,6 +22,19 @@ namespace conv2d {
 using OutputHeight = uint32_t;
 using OutputWidth = uint32_t;
 using Result = std::tuple<ttnn::Tensor, OutputHeight, OutputWidth, ttnn::Tensor, std::optional<ttnn::Tensor>>;
+using ResultWithOptions = std::variant<
+    ttnn::Tensor,
+    std::tuple<ttnn::Tensor, std::tuple<OutputHeight, OutputWidth>>,
+    std::tuple<ttnn::Tensor, std::tuple<ttnn::Tensor, std::optional<ttnn::Tensor>>>,
+    std::tuple<
+        ttnn::Tensor,
+        std::tuple<OutputHeight, OutputWidth>,
+        std::tuple<ttnn::Tensor, std::optional<ttnn::Tensor>>>>;
```

Comment:
This is getting to the point where an aggregate type with named data members might be more manageable than this.